lib/controller.ex

```diff
@@ -64,8 +64,13 @@ defmodule Controller do
     %{context | content: content, view: :index}
   end
 
-  def fetch_content(%{input: _, view: :grocery_list} = context) do
-    fetch_unknown_error(context)
+  def fetch_content(%{input: input, view: :grocery_list} = context) do
+    if RecipeParser.is_quantity(input) && input > 0 do
```

Comment:
This if statement could be a named validator function like `is_valid_quantity` that includes both the check both that it is an integer and that it is positive.